### PR TITLE
port spec 2

### PR DESCRIPTION
**Pull request by vadimeisenbergibm**
Resubmitting #235.

### CODE REVIEW

File: routing/v1alpha1/route_rule.proto [hunk 1]
```diff
@@ -493,3 +493,15 @@ message CorsPolicy {
   // Access-Control-Allow-Credentials header.
   google.protobuf.BoolValue allow_credentials = 6;
 }
+
+// Port describes the properties of a specific port of a service.
+message Port {
+  // A valid non-negative integer port number.
+  int32 number = 1;
+  // The protocol to communicate with the external services.
```
Comment (rshriram) [inline routing/v1alpha1/route_rule.proto]:
```oneof pspec {
  int32 number = 1;
  string name = 2;
}
string protocol = 3;```

Comment (vadimeisenbergibm) [reply routing/v1alpha1/route_rule.proto]: Done.

**Comment by rshriram** [comment]
Before you merge, make sure that this works with istio/istio..

File: routing/v1alpha1/route_rule.proto [hunk 2]
```diff
@@ -493,3 +493,17 @@ message CorsPolicy {
...
+     // A valid non-negative integer port number.
+    int32 number = 1;
+
+    // Name assigned to the port.
+    string name = 2;
+  }
+  
+  // MUST BE one of HTTP|HTTPS|GRPC|HTTP2|MONGO|REDIS|TCP.
+  string protocol = 3;
+}
```
Comment (rshriram) [inline routing/v1alpha1/route_rule.proto]: I forgot.. you have not fixed the documentation..
Also, why are we moving the Port out of egress Rule? is this for a specific use case? I did it in an attempt to refactor. I vaguely remember needing it for something else related to egress, but I can't recall what it is.

Comment (vadimeisenbergibm) [reply routing/v1alpha1/route_rule.proto]: This is just resubmission of your PR #235, I guess it is just for refactoring, I am not aware of any specific use case. I am fine with leaving egress rule as is. Up to you to decide.

Comment (rshriram) [reply routing/v1alpha1/route_rule.proto]: Yeah, in that case, let us wait. Once the new route rules settle down, we should have more clarity.

Comment (vadimeisenbergibm) [reply routing/v1alpha1/route_rule.proto]: Perfectly fine with me, closing the PR.

**Comment by vadimeisenbergibm** [comment]
Closed, decided to wait with this refactoring until more clarity achieved.

**Pull request closed without merging**